eeraster.py

```diff
@@ -0,0 +1,130 @@
+
+The `value_at` method returns the value of the raster at the specified
+latitude and longitude.
+"""
+class eeRaster:
+    """
+    A class representing an Earth Engine raster.
+
+    Args:
+        imageCollection (ee.ImageCollection): The Earth Engine ImageCollection
+        that represents the raster. This should be a single band normally
+        specified as:
+        ee.ImageCollection('projects/<project>/<collection>').select("<band>")
+    """
+    def __init__(self, imageCollection: ee.ImageCollection):
+      self._imageCollection = imageCollection
+
+    @staticmethod
+    def demXfab():
+      """
+      Returns an eeRaster representing digital elevation minus forest and
+      buildings. See https://gee-community-catalog.org/projects/fabdem/
+      """
+      global _demXfab
+      eeddf.initialize_ddf()
+      if (_demXfab is None):
+        _demXfab = eeRaster(ee.ImageCollection(
+          'projects/sat-io/open-datasets/FABDEM').select("b1"))
+      return _demXfab
+
+    @staticmethod
+    def dem():
+      """
+      Returns an eeRaster representing digital elevation (copernicus).
+      See https://gee-community-catalog.org/projects/glo30/
+      """
+      eeddf.initialize_ddf()
+      global _dem
+      if (_dem is None):
+        _dem = eeRaster(ee.ImageCollection(
+          'projects/sat-io/open-datasets/GLO-30').select("b1"))
+      return _dem
+
+    def value_at(self, lat: float, lon: float) -> float:
+        """
+        Returns the value of the raster at the specified latitude and longitude.
+
+        Args:
+            lat (float): The latitude coordinate.
+            lon (float): The longitude coordinate.
+
+        Returns:
+            float: The value of the raster at the specified latitude and longitude.
+        """
+        return self.values_at([[lat, lon]])[0]
+
+    def values_at(self, coordinates:List[Tuple[float, float]]) -> List[float]:
+      """
+      Returns the values of the raster at the specified latitudes and longitudes.
+
+      Args:
+          coordinates: A list of tuples consisting of [latitude, longitude].
+
+      Returns:
+          List[float]: The values of the raster at the specified latitudes and
+          longitudes.
+      """
+      df = self.values_at_df(pd.DataFrame(coordinates, columns=[
+        _LATITUDE_COLUMN_NAME, _LONGITUDE_COLUMN_NAME]))
+
+      value_list = df["value"].tolist()
+      return value_list
+
+    def values_at_df(self, coordinates:pd.DataFrame, column_name: str = "value") -> pd.DataFrame:
+      """
+      Returns the values of the raster at the specified latitudes and longitudes.
+
+      Args:
+          coordinates: A list of tuples consisting of [latitude, longitude].
+
+      Returns:
+          List[float]: The values of the raster at the specified latitudes and
+          longitudes.
+      """
+      coordinates[_LATITUDE_COLUMN_NAME] = coordinates[_LATITUDE_COLUMN_NAME].astype(float).round(5)
```

Comment:
Is there a specific reason we're rounding to 5 decimals here?

Reply:
yes, earth engine does not give back the exact lat/lon and we need to join it with the original dataframe. It may be a float epsilon issue.
rounding to 5 decimals rounds to 11 meters which should preserve the precision we need.

Reply:
This would be really helpful as a comment ;-)

Reply:
added!